server/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -328,7 +328,13 @@ private void tryAllHandlers(final RestRequest request, final RestChannel channel
                       return;
                   }
                 } else {
-                    dispatchRequest(request, channel, handler);
+                    if(handler.compatibilityRequired() == false //regular (not removed) handlers are always dispatched
+                        //handlers that were registered compatible, require request to be compatible
+                        || CompatibleHandlers.isV7Compatible(request)) {
```

Comment:
any way to remove the v7 here ? ideally this would be agnostic to the actual version (which I think it already is in all but the name)

Reply:
good point, the generic and always present compatible code can be made agnostic.